### PR TITLE
Back end signup / login api routes

### PR DESCRIPTION
**Pull request by quddd**
Initial draft for BE LogIn / SignUp API

### CODE REVIEW

**Comment by quddd** [comment]
Need to resolve the following isssue:
(node:15944) DeprecationWarning: collection.ensureIndex is deprecated. Use createIndexes instead.

File: README.md [hunk 1]
```diff
@@ -14,3 +14,18 @@ In the `.env` file create the following two variables and assign them the mongoU
     DATABASE_PASSWORD=<db_password_goes_here>
 
 You should now be able to connect to the mongoDB database.
+
+# In the `.env` file create the TOKEN_SECRET variable
+
+TOKEN_SECRET=<token_secret>
+
+Finished Bacjend Login / Register API routes
```
Comment (dproc96) [inline README.md]:
```suggestion
Finished Backend Login / Register API routes
```

File: server/routes/auth.js [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+        const validPassword = await bcrypt.compare(password, user.password);
+        if(!validPassword){
+            return res.status(400).send('Invalid password');
+        }
+
+        //create and assign a token
+        const token = jwt.sign({_id: email}, process.env.TOKEN_SECRET);
+        //create httpOnly cookie
+        res.cookie("jwt_token", token, {
+            maxAge: 3600, // 1 hour
```
Comment (dproc96) [inline server/routes/auth.js]: This is a fairly short use cookie, that's up to you but a user might expect the cookie to last longer.

**Review by dproc96 — COMMENTED** [review]
Only had a couple of comments on your code, but there's one thing you should add to this PR: registering should also return a token, logging the user in as part of the registering process.

**Review by dproc96 — APPROVED** [review]
This looks good! Just be careful with the .env when you fix the conflicts merge this.

**Pull request merged**